src/main/java/ui/MenuControl.java

```diff
@@ -346,4 +348,22 @@ private void updateRepoRemoveList(Menu remove) {
     private void onRepoRemove(String repoId) {
         ui.logic.removeStoredRepository(repoId).thenRun(() -> ui.triggerEvent(new UnusedStoredReposChangedEvent()));
     }
+
+    private Menu createUpdateMenu() {
+        Menu update = new Menu("Update");
+
+        MenuItem checkProgress = new MenuItem("Check progress...");
+
+        if (ui.updateManager == null) {
```

Comment:
Is it possible to stub the update manager out in test mode instead of not creating it?